pkg/util/cache.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"sync"
+	"time"
+)
+
+// CacheItem wraps T so it has a LastUpdated field and can be used with Cache.
+type CacheItem[T any] struct {
+	// Item is the cached item.
+	Item T
+
+	// LastUpdated is the last time the item was updated.
+	LastUpdated time.Time
+}
+
+// Cache is a generic implementation of a cache that can be configured with a
+// maximum number of items and can evict items after a certain amount of time.
+type Cache[T any] struct {
+	mu       sync.Mutex
+	items    map[string]CacheItem[T]
+	maxItems int
+	done     chan struct{}
+	donce    sync.Once
+
+	// expiredChan returns a channel on which the IDs of items that are expired
+	// are sent. This channel is closed when the cache is closed, but only after
+	// all, pending eviction notices are received.
+	expiredChan chan string
+	expiringNum sync.WaitGroup
+}
+
+// NewCache initializes a new cache with the provided expiration options.
```

Comment:
It would be helpful to document the cache behavior with regarding to the expireAfter and maxItems, also LastUpdated in CacheItem.

Reply:
I think those are already documented in the `Cache` and `CacheItem` structs. Will keep the current comments for now as they are all in the same file.